SportsHub/SportsHub.Api/Exceptions/ExceptionHandler.cs

```diff
@@ -1,20 +1,55 @@
 ﻿using System.Net;
+using SportsHub.Api.Exceptions.CustomExceptionModels;
 
 namespace SportsHub.Api.Exceptions;
 
 public class ExceptionHandler: IMiddleware
 {
+    private CustomException _exception;
     public async Task InvokeAsync(HttpContext context, RequestDelegate next)
     {
         try
         {
             await next(context);
         }
-
-        catch (Exception e)
+        catch (BusinessLogicException ex)
+        {
```

Comment:
you can use "When" here

Reply:
I totaly forgot about this keyword . I will improve it